flow/project.py

```diff
@@ -3773,24 +3772,48 @@ def _gather_flow_groups(self, names=None):
 
         """
         if names is None:
-            # If no names are selected, use all singleton groups
-            operations = [self._groups[name] for name in self.operations]
-        else:
-            operations = {}
-            for name in names:
-                if name in operations:
-                    continue
-                groups = [
-                    group
-                    for group_name, group in self.groups.items()
-                    if re.fullmatch(name, group_name)
-                ]
-                if len(groups) > 0:
-                    for group in groups:
-                        operations[group.name] = group
-                else:
-                    continue
-            operations = list(operations.values())
+            return list(self._groups.values())
+        operations = {}
+        for name in names:
+            if name in operations:
+                continue
+            groups = [
+                group
+                for group_name, group in self.groups.items()
+                if re.fullmatch(name, group_name)
+            ]
+            for group in groups:
+                operations[group.name] = group
+        return list(operations.values())
+
+    def _gather_executable_flow_groups(self, names=None):
+        r"""Grabs immediately executable flow groups that match any given name.
+
+        The provided names can be any regular expression that fully match a group name.
+
+        Note
+        ----
+            The behavior is distinct from ``_gather_selected_flow_groups`` in that
+            for execution the default set is not all groups but all singelton
+            groups (operations).
```

Comment:
Don't indent this note.
```suggestion
        The behavior is distinct from ``_gather_selected_flow_groups`` in that
        for execution the default set is not all groups but all singleton
        groups (operations).
```